### PR TITLE
Adding Support For Staging Deployment and Test

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: .github/workflows/production_deployment.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true
+      - name: push to dockerhub
+        uses: fylein/docker-release-action@master
+        env:
+          DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+          DOCKERHUB_PASSWORD: ${{ secrets.DOCKERHUB_PASSWORD }}
+          IMAGE_NAME: fyle_admin_settings-api
```
Comment (Shwetabhk) [inline .github/workflows/production_deployment.yml:20]: image name is not correct
`fyle_integrations-settings-api`

File: .github/workflows/production_deployment.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+          KUBERNETES_MANIFEST_FILE_PATH: deployment/production/controller.yml
+      - name: Slack Notification
+        uses: rtCamp/action-slack-notify@v2
+        env:
+          SLACK_COLOR: ${{ job.status }}
+          SLACK_MESSAGE: Production Deploy Status - ${{ job.status }}
+          SLACK_TITLE: Deploy Partner Dashboard API to Production
```
Comment (Shwetabhk) [inline .github/workflows/production_deployment.yml:40]: Fix title

File: .github/workflows/staging_deployment.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true
+      - name: push to dockerhub
+        uses: fylein/docker-release-action@master
+        env:
+          DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+          DOCKERHUB_PASSWORD: ${{ secrets.DOCKERHUB_PASSWORD }}
+          IMAGE_NAME: fyle_admin_settings-api
```
Comment (Shwetabhk) [inline .github/workflows/staging_deployment.yml:24]: `fyle_integrations-settings-api`

File: .github/workflows/staging_deployment.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+          KUBERNETES_MANIFEST_FILE_PATH: deployment/staging/controller.yml
+      - name: Slack Notification
+        uses: rtCamp/action-slack-notify@v2
+        env:
+          SLACK_COLOR: ${{ job.status }}
+          SLACK_MESSAGE: Staging Deploy Status - ${{ job.status }}
+          SLACK_TITLE: Deploy Admin Settings API to Staging
```
Comment (Shwetabhk) [inline .github/workflows/staging_deployment.yml:44]: Integrations Settings

File: controller/staging/controller.yml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            - containerPort: 8000
+          readinessProbe:
+            httpGet:
+              path: /api/orgs/ready/
+              port: 8000
+            initialDelaySeconds: 10
+            timeoutSeconds: 1
+          envFrom:
+            - secretRef:
+                name: integrations_settings-api-secrets
```
Comment (Shwetabhk) [inline controller/staging/controller.yml:55]: always use `-` for delpoy files dont use underscores

File: controller/staging/controller.yml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  replicas: 1
+  selector:
+    matchLabels:
+      app: integrations_settings-api
+  strategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: integrations_settings-api
```
Comment (Shwetabhk) [inline controller/staging/controller.yml:32]: integrations-settings-api

File: controller/staging/controller.yml [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+metadata:
+  name: integrations_settings-api
+  namespace: integrations
+  labels:
+    app: integrations_settings-api
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: integrations_settings-api
```
Comment (Shwetabhk) [inline controller/staging/controller.yml:26]: integrations-settings-api

File: controller/staging/controller.yml [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    app: integrations_settings-api
+  type: ClusterIP
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: integrations_settings-api
+  namespace: integrations
+  labels:
+    app: integrations_settings-api
```
Comment (Shwetabhk) [inline controller/staging/controller.yml:21]: integrations-settings-api

File: controller/staging/controller.yml [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  ports:
+    - port: 8000
+  selector:
+    app: integrations_settings-api
+  type: ClusterIP
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: integrations_settings-api
```
Comment (Shwetabhk) [inline controller/staging/controller.yml:18]: integrations-settings-api

File: controller/staging/controller.yml [hunk 10]
```diff
@@ -0,0 +1,57 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: integrations_settings-api
+  namespace: integrations
+  labels:
+    app: integrations_settings-api
```
Comment (Shwetabhk) [inline controller/staging/controller.yml:7]: integrations-settings-api

File: controller/staging/controller.yml [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+spec:
+  ports:
+    - port: 8000
+  selector:
+    app: integrations_settings-api
```
Comment (Shwetabhk) [inline controller/staging/controller.yml:12]: integrations-settings-api

File: controller/staging/controller.yml [hunk 12]
```diff
@@ -0,0 +1,57 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: integrations_settings-api
```
Comment (Shwetabhk) [inline controller/staging/controller.yml:4]: integrations-settings-api

File: controller/production/controller.yml [hunk 13]
```diff
@@ -0,0 +1,57 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: integrations_settings-api
```
Comment (Shwetabhk) [inline controller/production/controller.yml:4]: Same comments as staging file

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**